x-pack/plugins/cases/server/connectors/cases/utils.ts

```diff
@@ -48,3 +51,30 @@ export const convertValueToString = (value: unknown): string => {
 
   return toString(value);
 };
+
+export const buildRequiredCustomFieldsForRequest = (
+  customFieldsConfiguration?: CustomFieldsConfiguration
+): CaseRequestCustomFields => {
+  // only populate with the default value required custom fields missing from the request
+  return customFieldsConfiguration
+    ? customFieldsConfiguration
+        .filter((customFieldConfig) => customFieldConfig.required)
+        .map((customFieldConfig) => {
+          let value = null;
+
+          if (customFieldConfig.type in VALUES_FOR_CUSTOM_FIELDS_MISSING_DEFAULTS) {
+            value =
+              customFieldConfig.defaultValue === undefined ||
+              customFieldConfig?.defaultValue === null
+                ? VALUES_FOR_CUSTOM_FIELDS_MISSING_DEFAULTS[customFieldConfig.type]
+                : customFieldConfig?.defaultValue;
+          }
+
+          return {
+            key: customFieldConfig.key,
+            type: customFieldConfig.type,
+            value: value as CaseRequestCustomField['value'],
+          } as CaseRequestCustomField;
```

Comment:
Now that you cast the value this is not needed, right?

Reply:
No, that still gives the type error I was avoiding.

Reply:
That's a pitty. Then what I said about casting the value does not make sense. I thought it could work.